lte/gateway/c/core/oai/tasks/mme_app/mme_app_bearer.c

```diff
@@ -966,6 +966,33 @@ void mme_app_handle_delete_session_rsp(
     OAILOG_FUNC_OUT(LOG_MME_APP);
   }
 
+  if (ue_context_p->tau_accept_eps_ber_cntx_status) {
+    ue_context_p->nb_delete_sessions--;
+    OAILOG_DEBUG_UE(
+        LOG_MME_APP, ue_context_p->emm_context._imsi64,
+        "Freeing the bearer context as tau_accept_eps_ber_cntx_status flag is "
+        "set for "
+        "ue_id " MME_UE_S1AP_ID_FMT " and lbi=%u\n",
+        ue_context_p->mme_ue_s1ap_id, delete_sess_resp_pP->lbi);
+
+    int bearer_idx = EBI_TO_INDEX(delete_sess_resp_pP->lbi);
+    eps_bearer_release(
+        &ue_context_p->emm_context, delete_sess_resp_pP->lbi, &pid,
+        &bearer_idx);
+    if (ue_context_p->pdn_contexts[pid]) {
+      mme_app_free_pdn_context(
+          &ue_context_p->pdn_contexts[pid], ue_context_p->emm_context._imsi64);
+    }
+    // Free bearer context entry
+    if (ue_context_p->bearer_contexts[bearer_idx]) {
+      free_wrapper((void**) &ue_context_p->bearer_contexts[bearer_idx]);
```

Comment:
ue_context_p->bearer_contexts[bearer_idx] is not a pointer, what are you trying to free here? TFTs, PCOs, etc. need to be freed (which are done above?).

Reply:
ue_context_p->bearer_contexts is a list of pointers for which memory gets allocated in mme_app_create_bearer_context(). Hence it needs to be freed. TFT and PCO are freed before, in eps_bearer_release()